builtin/rebase.c

```diff
@@ -52,6 +57,13 @@ struct rebase_options {
 	int preserve_merges;
 	const char *upstream_name;
 	const char *state_dir;
+	struct commit *upstream;
```

Comment:
Most of these fields are grouped sensibly, more or less. But `upstream` should "cuddle" with `upstream_name`.

In general, I would like to keep the fields in groups that make some kind of sense, like: internal fields, i.e. fields with settings that will not be handed down to the rebase backend via the `env` array first (right now, there is only `type`). Then the variables that will be handed to the backend, maybe in some sort of order (intuitively, I would put `state_dir` first because it is kinda real low-level, then the revision-related variables, then options such as `root`).

What do you think @prertik?

Reply:
Maybe this can wait?

Reply:
Sure. As long as the `preserve_merges` field is removed from the patch that introduced it, before the patches hit the Git mailing list, I'm cool ;-)